pkg/backup/backupinfo/manifest_handling.go

```diff
@@ -875,6 +875,60 @@ func GetLocalityInfo(
 	return info, nil
 }
 
+type backupTreeEntry struct {
```

Comment:
I think we should make this struct public so we can use it as part of the manifest APIs. Which would allow ValidateEndTimeAndTruncate to return the struct and eventually allow us to get rid of the zip and unzip logic.